### PR TITLE
Create targets group to caclulate POI for all US/US territory lake

### PR DESCRIPTION
**Pull request by steeleb**
Heya MB!

First PR for lakeSR coming at you! This looks big, but most of it (6.5k lines) is the rendered run_targets.html. As a warning, this group takes legit 4h to run, likely faster if you're plugged into ethernet (which I am not) because there is a loooot of downloading that happens in this targets group. No OneDrive connections, no .Renviron vars.

This PR is just about calculating the point of inaccessibility (aka "Chebyshev Center", fka "deepest point"). The actual calculation of POI is copied from the Landsat_CT_SRST workflow - and yes, it still has that silly for-loop in it. 

Re the for-loop in `a_Calculate_Centers/src/calculate_AK_poi.R`:  I'll be honest I spent a fair amount of time trying to work up an alternative function for that, but my brain could nooooot figure it out. SO! If your brain wraps around the problem better than mine (which I wouldn't doubt because your function writing ability is wellll beyond mine), I'd welcome alternatives. Also, it's functional and doesn't need to be repeated frequently, so NBD either way.

The resulting centers file contains nearly 1.5 million lake centers. :)

Future dev notes: I'll be adding a bookdown to this in a future PR to add the literate side of things for this workflow. 

Thanks in advance!

### CODE REVIEW

**Comment by steeleb** [comment]
@kathryn-willi As an FYI for how I handled AK - I dropped all VPUIDs related to AK and then downloaded the whole dang state of AK Best Res .zip from The National Map and then calculated centers from that. Fun fact: there are SO MANY lakes in AK. ~450k are >1ha and lakes/res/impoundments. 🤯 There are about 1.5M lakes in our centers dataset.

File: a_Calculate_Centers/src/calculate_centers_HUC4.R [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    
+    # filter the waterbodies for ftypes of interest. 390 = lake/pond; 436 = res;
+    # 361 = playa
+    wbd <- wbd %>% 
+      filter(FTYPE %in% c(390, 436, 361)) 
+    
+    # filter for area > 1 hectare (0.01 km^2)
+    wbd <- wbd %>% 
+      filter(AreaSqKM >= 0.01) 
+
```
Comment (mbrousil) [inline a_Calculate_Centers/src/calculate_centers_HUC4.R:43]: suggestion: You could condense these lines here, not a huge deal but more concise

```suggestion
    wbd <- wbd %>% 
         filter(
        # filter the waterbodies for ftypes of interest. 390 = lake/pond; 436 = res;
        # 361 = playa 
             FTYPE %in% c(390, 436, 361),
       # ...and for area > 1 hectare (0.01 km^2)
             AreaSqKM >= 0.01) 
```

File: a_Calculate_Centers/src/calculate_centers_HUC4.R [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      filter(grepl("SURFACE", st_geometry_type(Shape)))
+    if (nrow(multisurface_wbd) > 0) {
+      st_write(multisurface_wbd, file.path("a_Calculate_Centers/multisurface/",
+                                          paste0("dropped_multisurface_geo_",
+                                                 HUC4, ".gpkg")))
+    }
+    
+    # check for valid geometry and drop z coords (if they exist)
+    wbd <- wbd %>% 
+      # there are a few weirdos that are multisurface geometires that {sf} doesn't
```
Comment (mbrousil) [inline a_Calculate_Centers/src/calculate_centers_HUC4.R:56]:
```suggestion
      # there are a few weirdos that are multisurface geometries that {sf} doesn't
```

File: a_Calculate_Centers.R [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  tar_target(
+    name = NHDHR_poi_points_noAK,
+    command = {
+      #make a list of incomplete VPUIDs
+      incomplete_list <- unique(NHDHR_poi_points$VPUID) %>% 
+        # there are a few HUC8s in Alaska with data that we want to drop, these
+        # are indicated by VPUIDs that are 8 characters - and there are a few that
+        # were concatenated in a previous step, resulting in 18 characters. All 
+        # other VPUIDs are either 4, 5, 6, 10, or 11 characters in length.
+        .[nchar(.)==8 | nchar(.) == 18]
```
Comment (mbrousil) [inline a_Calculate_Centers.R:124]:
```suggestion
        .[nchar(.) == 8 | nchar(.) == 18]
```

File: a_Calculate_Centers/src/calculate_AK_poi.R [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+#' calculate POI for each polygon
+#' 
+#' @returns silently saves .csv file in mid folder of the POI centers and 
+#' associated WBD metadata
+#' 
+#' 
+calculate_AK_poi <- function() {
+  # set timeout so that... this doesn't timeout
+  options(timeout = 60000)
+  # url for the NHD Best Resolutoin for
```
Comment (mbrousil) [inline a_Calculate_Centers/src/calculate_AK_poi.R:16]:
```suggestion
  # url for the NHD Best Resolution for 
```

File: a_Calculate_Centers/src/calculate_AK_poi.R [hunk 5]
```diff
@@ -0,0 +1,129 @@
+#' @title Calcuate POI center for NHD Best Resolution waterbodies from The
```
Comment (mbrousil) [inline a_Calculate_Centers/src/calculate_AK_poi.R:1]: question: Should the title here mention AK?

File: a_Calculate_Centers/src/calculate_AK_poi.R [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+                 layer = 'NHDWaterbody')
+  
+  # filter the waterbodies for ftypes of interest. 390 = lake/pond; 436 = res;
+  # 361 = playa
+  wbd <- wbd %>% 
+    filter(ftype %in% c(390, 436, 361)) 
+  
+  # filter for area > 1 hectare (0.01 km^2)
+  wbd <- wbd %>% 
+    filter(areasqkm >= 0.01)
```
Comment (mbrousil) [inline a_Calculate_Centers/src/calculate_AK_poi.R:36]: suggestion: Same as in the other function

```suggestion
 wbd <- wbd %>% 
  filter(
    # filter the waterbodies for ftypes of interest. 390 = lake/pond; 436 = res;
    # 361 = playa
    ftype %in% c(390, 436, 361),
    # ...and for area > 1 hectare (0.01 km^2)
    areasqkm >= 0.01) 
```

**Comment by mbrousil** [comment]
Hey @steeleb this looks good to me. No major comments. Would you like me to run it locally? I wasn't sure if that was your intent or not.

**Comment by steeleb** [comment]
Rockstar, thank you @mbrousil!

I don' think there is a need to run right now, I basically ran the whole dang thing shortly before this PR. I think, like with AquaMatch, we'll do one big run when we're ready to have external eyes? I don't have strong feelings at this juncture!

File: a_Calculate_Centers/src/calculate_AK_poi.R (hunk 5 above)
Comment (steeleb) [reply a_Calculate_Centers/src/calculate_AK_poi.R:1]: It absolutely should!

**Comment by steeleb** [comment]
Thanks for reviewing, MB. I accepted all of the suggestions and added text for that Alaska function.

Also, thanks for finding all of my dyslexic spelling errors. ha! (But seriously, thank you.)

Am I good to merge from your perspective?

**Comment by mbrousil** [comment]
Yep!

**Pull request merged**